logstash-core/lib/logstash/plugin.rb

```diff
@@ -31,9 +49,21 @@ class LogStash::Plugin
   # }
   # ```
   #
-  # If you don't explicitely set this variable Logstash will generate a unique name.
+  # If you don't explicitly set this variable Logstash will generate a unique name.
   config :id, :validate => :string
 
+  # Add a set of specific operation timing thresholds for the plugin, they are used in the context
+  # of reporting slow operations.
+  #
+  # ```
+  #  thresholds => {
+  #    filter.grok.runtime => 12
+  #  }
+  # ```
+  #
+  # The values defined here takes precedence over the ones defined in the logstash config file. (default => {})
+  config :thresholds, :validate => :hash, :default => {}
```

Comment:
Wise comments @jordansissel, the situation why I chose to implement this that way is because in #5733 we agreed to have thresholds like:

* In the settings file.
* Per plugin, taking precedence over the settings.

So I made the assumption all plugins should have this config option available, If I understand you properly, are you suggesting to have this option only defined in plugins that have and use the slow log?

re the usage of hash, my problem here is I see the thresholds as actually a hash, as you can see in the documentation. What are you thinking as alternate structure?